main.go

```diff
@@ -130,9 +167,41 @@ var webSocketUpgrade = websocket.Upgrader{
 	},
 }
 
+func validateTokenMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		tokenString := c.GetHeader("Authorization")
+
+		if tokenString == "" {
+			c.JSON(http.StatusUnprocessableEntity, gin.H{"err": "Token is required"})
+			c.Abort();
+		}
+
+		// get token without 'Bearer ' in name
+		tokenString = tokenString[len("Bearer "):]
+
+		token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
+			return verifyKey, nil
+		})
+
+		// if there are any errors or token is invalid, abort and don't 
+		// execute the next function in the chain
+		if err != nil {
+			c.JSON(http.StatusUnauthorized, gin.H{"err": err.Error()})
+			c.Abort()
+			return
+		}
+		if !token.Valid {
+			c.JSON(http.StatusUnauthorized, gin.H{"err": err.Error()})
+			c.Abort()
+			return
+		}
+	}
+}
+
 func main() {
 	d := database.Database{}
 	db = &d
+	initKeys()
```

Comment:
If this needs to be run at startup every time, no matter what, maybe take a look at adding an init() function:

[When is the init function ran](https://stackoverflow.com/questions/24790175/when-is-the-init-function-run)

This kinda cedes some of the control over to go in terms of lifecycle but it seems fine given it's all setup.